core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java

```diff
@@ -146,4 +141,38 @@ public void authenticate(String connectionUser, String impersonationUser)
         connectionUser, impersonationUser, connectionUser, impersonationUser,
         RuntimeConstants.ALLUXIO_SECURITY_DOCS_URL));
   }
+
+  @Override
+  public void update() {
+    loadImpersonationUser(Configuration.global());
+  }
+
+  private void loadImpersonationUser(AlluxioConfiguration conf) {
+    Map<String, Set<String>> impersonationGroups = new HashMap<>();
+    Map<String, Set<String>> impersonationUsers = new HashMap<>();
+    for (PropertyKey key : conf.keySet()) {
```

Comment:
Just concern to the effect of old implementation, we have to iterate all PropertyKey to find the template property, although the updated propertykeys are not them.

Reply:
Currently, there are approximately 1000 property keys in Alluxio. Traversing them does not significantly affect performance. However, it would be more effective to operate based on the PropertyKey that changes with modifications.

Reply:
For current implementation, we can have two idea to reduce waste loop templete key improve the performance
- Improve the update configuration and reconfig framework, support fire change and listen changed properties only.
- Improve the propertykey framework to support index template keys, so we can get the template key efficiently.